src/3ds.rs

```diff
@@ -7,10 +7,12 @@
 // except according to those terms.
 
 //! Implementation for Nintendo 3DS
+use core::mem::MaybeUninit;
```

Comment:
Note: This increases our MSRV to 1.36.0 so is a breaking change. However, if we gate everything behind the `read_buf` feature, this shouldn't be an issue.